app/src/ui/app.tsx

```diff
@@ -395,8 +395,6 @@ export class App extends React.Component<IAppProps, IAppState> {
       case 'find-text':
         return this.findText()
     }
-
-    return assertNever(name, `Unknown menu event name: ${name}`)
```

Comment:
We used `assertNever` here not only to catch errors whenever the switch statement did not handle all potential values of `name` but also to throw a runtime error in the case we receive an event with an unknown `name`.

I see that now with the upgraded TypeScript we're going to be able to catch the first issue but do you know if there's a way to keep these runtime checks (or similar ones) without TypeScript complaining about unreachable code and without setting the global `allowUnreachableCode` config to true?

Reply:
I'm pretty sure there is - let me get back to you.

Reply:
There is, but it looks awfully similar to what you're doing now: https://stackoverflow.com/questions/39419170/how-do-i-check-that-a-switch-block-is-exhaustive-in-typescript

Maybe I misread the error and it was actually saying you were missing a case?

Reply:
I can't say that I find this answer particularly satisfying, but it works if you [move ](https://www.typescriptlang.org/play/#code/KYOwrgtgBAwg9gGzgJygbwLACgq6gJWABMAabPKAcWWFDJzwCEExhsBfbbAMzBAGMALgEs4IKAEMAzlODJBAVRA0J-ABYSARgmAAKAB4AuKCGAA3OQEpjpi6kwNcgtcjgB3E8A8BRZK+S6AEQAIsJEIADkglDA+gAOwEJQgnBQAObA0WpywIGWANwcXFi8AiJi6ZnwSMgAchIQevzG1SjWUFKCyMIgaejkeFJuwoLquvyW-Y4UUPzSwLCIKAB0hESGAzMzNIJgyOIRNEQRhdMUc7KLNcvUtCAbZ1tQO3sHaTSgJ5szFwutyMtmKwHk9tplXlAItpWF9HrgiMBuBIwAhBCDQXhpLJ5EoVOotDpxgVvpwsOwgA)the assert into the `default` case.  That was probably the only way we could distinguish switch exhaustiveness checks from actually unreachable code.

Reply:
I've restored the `assertNever`s but am holding off on pushing until I figure out how to make prettier happy.